Graph algo

Suxiaocai opened the pull request:
1. Fixed PageRank template
2. Added shortest path algos
3. Added similarity algos
4. Added install free queries

xchang2014 commented:
1. Please add Jaccard to the install.sh list. Only asking for graph name is fine.

Suxiaocai commented:
1. Added Jaccard as gtmp, using same format as cosine similarity
2. Added reverse edge choice for similarity algos 
3. Added uniform weight for cosine similarity
4. Added loop to let user try again when they type wrong edge attribute name

xchang2014 commented on graph_algorithms/algorithms/install.sh:
```diff
@@ -153,130 +184,211 @@ while [ !$finished ]; do
+
+	# 4.2 Ask for reverse edge type for similarity algos. 
+        if [[ ${algoName} == similarity* ]]; then
+		read -p 'Reverse Edge type: ' reveg
```
Should be second hop edge type. not necessarily a reverse edge， right?

xchang2014 commented on graph_algorithms/algorithms/generated/shortest_ss_pos_wt_file.gsql:
```diff
@@ -0,0 +1,75 @@
+# Algorithm for Single-Source Shortest Path with positive weights on edges 
+# The path parameter let user choose whether calculate path or not
+
+CREATE QUERY shortest_ss_pos_wt_file (VERTEX v, BOOL display, BOOL path, STRING filepath) FOR GRAPH social {
```
remove display, path. Print the whole path by default.

Suxiaocai replied:
Shall l totally remove display for every query? There is also display for queries like Closeness Centrality. Or just keep the display choice for ACCUM version?

Suxiaocai commented on graph_algorithms/install_free_algos/example/similarity_jaccard_single_attr.gsql:
```diff
@@ -0,0 +1,30 @@
+  Start (ANY) = {input};
+    Start = select s
+            from Start:s
+            accum @@set_size_A += s.outdegree(edge_type);
```
yes, it works
